packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -753,18 +753,33 @@ class TextSelectionOverlay {
       renderEditable: renderObject,
     ));
 
-    _handleSelectionHandleChanged(newSelection, isEnd: false);
+    _handleSelectionHandleChanged(newSelection, handleToView: TextSelectionHandleSide.baseHandle);
   }
 
   void _handleAnyDragEnd(DragEndDetails details) => _selectionOverlay.hideMagnifier(shouldShowToolbar: !_selection.isCollapsed);
 
-  void _handleSelectionHandleChanged(TextSelection newSelection, {required bool isEnd}) {
-    final TextPosition textPosition = isEnd ? newSelection.extent : newSelection.base;
+  void _handleSelectionHandleChanged(TextSelection newSelection, {required TextSelectionHandleSide handleToView}) {
+    final TextPosition positionToView;
+    switch (defaultTargetPlatform) {
+      case TargetPlatform.iOS:
+      case TargetPlatform.macOS:
+        // On iOS and macOS, extent is always the dragged handle.
+        // (https://github.com/flutter/flutter/pull/105069)
+        positionToView = newSelection.extent;
+        break;
+      case TargetPlatform.android:
+      case TargetPlatform.fuchsia:
+      case TargetPlatform.linux:
+      case TargetPlatform.windows:
+        positionToView = handleToView == TextSelectionHandleSide.extentHandle ? newSelection.extent : newSelection.base;
+        break;
+    }
     selectionDelegate.userUpdateTextEditingValue(
       _value.copyWith(selection: newSelection),
       SelectionChangedCause.drag,
+      handleToView: handleToView,
     );
-    selectionDelegate.bringIntoView(textPosition);
+    selectionDelegate.bringIntoView(positionToView);
```

Comment:
It looks like bringIntoView may also be called as a result of userUpdateTextEditingValue.  Would it be possible to avoid calling it twice like that?

I know that we were probably doing that even before your PR...

Reply:
I removed this line.